### PR TITLE
ES2015 and promisify

### LINKED ISSUE #8: ES2015 and promisify

### PR DESCRIPTION
**Pull request by kevva**
Used `got` instead of core `http`/`https` for convenience. Could switch back if it's too heavyweight.

Fixes #8.

### CODE REVIEW

**Comment by silverwind** [comment]
I'd actually suggest `request` for HTTP. It has the benefit of automatically respecting [environment-defined proxies](https://github.com/request/request#controlling-proxy-behaviour-using-environment-variables), and IIRC, the request for proxy support already came up in the past.

**Comment by silverwind** [comment]
Oh, and it could be done separately, but for https://github.com/sindresorhus/is-online/pull/28#issuecomment-245644992, we'd need a `timeout` options, which should map to both the `http` library's and the existing `is-port-reachable` option.

I was thinking about doing a PR for it, but with this rewrite lingering, I guess the efforts would be fruitless.

**Comment by kevva** [comment]
Supplying [caw](https://github.com/kevva/caw) as an `agent` option should cover proxies, *somewhat*. I'd be glad if someone could point out what's missing in that module as opposed to `request`. I just want to extract all the proxy logic so we could use proxies in harmony with `got`.

**Comment by silverwind** [comment]
Thanks, I wasn't aware of `caw` yet. Will test it later. I assume some additional logic will be needed to support the `NO_PROXY` environment variable. But let's not hold up this PR with that.

File: index.js [hunk 1]
```diff
@@ -1,69 +1,50 @@
...
+		}
+
+		return true;
+	});
+};
 
-		var host = dest.hostname;
-		var port = dest.port || 80;
+module.exports = dests => {
+	return Promise.all(arrify(dests).map(x => {
```
Comment (sindresorhus) [inline index.js]: This is a bit inefficient as it waits for all the dests to finish. We really only care about the fastest succeeding one. You could use [`p-any`](https://github.com/sindresorhus/p-any) for this.

Same with the browser version.

File: index.js [hunk 2]
```diff
@@ -1,69 +1,50 @@
...
-					} else {
-						done();
-					}
-				});
-			}
-		});
-	}, function () {
-		cb(null, false);
-	});
+				return pify(isPortReachable)(port, {host: address});
```
Comment (sindresorhus) [inline index.js]: This now has a promise interface so `pify` is moot. Just bump the dependency.

**Comment by sindresorhus** [comment]
@silverwind Good to merge?

**Comment by silverwind** [comment]
LGTM

**Comment by sindresorhus** [comment]
Thanks @kevva :)

**Pull request merged**